systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java

```diff
@@ -744,6 +748,120 @@ void testNetworkPoliciesWithTlsListener() throws Exception {
         assertThrows(AssertionError.class, () -> availabilityTest(50, CLUSTER_NAME, true, topic1, kafkaUser, kafkaClientsNewPodName));
     }
 
+    @Test
+    void testTlsHostnameVerificationWithKafkaConnect() {
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 3, 1).done();
+        LOGGER.info("Getting IP of the bootstrap service");
+
+        String ipOfBootstrapService = kubeClient().getService(KafkaResources.bootstrapServiceName(CLUSTER_NAME)).getSpec().getClusterIP();
+
+        LOGGER.info("Kafka connect without config {} will not connect to {}:9093", "ssl.endpoint.identification.algorithm", ipOfBootstrapService);
+
+        KafkaConnect kafkaConnect = testMethodResources().kafkaConnectWithoutWait(testMethodResources().defaultKafkaConnect(CLUSTER_NAME, 1)
+                .editSpec()
+                    .withNewTls()
+                        .addNewTrustedCertificate()
+                            .withSecretName(CLUSTER_NAME + "-cluster-ca-cert")
+                            .withCertificate("ca.crt")
+                        .endTrustedCertificate()
+                    .endTls()
+                    .withBootstrapServers(ipOfBootstrapService + ":9093")
+                .endSpec()
+                .build());
+
+        TestUtils.waitFor("Waiting till kafka connect status will be 'NotReady'", Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_RESOURCE_READINESS,
```

Comment:
Thsi is fairly expensive test which will wait for a whole one timeout interval. Is there no better way to test that the connection is failing? Do we want to just assume it will be failing without actually testing it?

Reply:
I tried way. where i will await `io.strimzi.TimeoutException` but i took 14mins (because our `TIMEOUT_FOR_RESOURCE_READINESS = 14m`. This approach takes 9m. Another way could be that i will check only status of the pod that is in `crashloopbackoff` and can add check that pod has more that 3 restarts. This will takes like 4m in the summary.

Reply:
So basically there is one approach what i have in mind. This scenario takes like 9 mins i also tried another one when i await on `io.strimzi.TimeoutException` but i took 14mins because of resource timeout. Possibly i can check only pod status that is in `CrashLoopBackOff` and also wait until pod will restart more than 3 times. What do you think?

Reply:
The status is set only at the end of the reocnciliation. So I think it should not matter whether you wait for TimeoutException or for NotReady status.